File: Python/7py 1/factorial.py
"""
|--------------------------------------------------------------------------
| Factorial
|--------------------------------------------------------------------------
|
"""

def factorial(n):
   if n < 0 or n > 12: raise ValueError
   return n * factorial(n-1) if n else 1


def factorial(n):
   if n not in range(0, 13):
      raise ValueError('Error')
   return n * factorial(n - 1) if n else 1


def factorial(n):
   if 0 <= n <= 12:
      return n * factorial(n-1) if n else 1
   raise ValueError


print(factorial(0), 1)
print(factorial(1), 1)
print(factorial(2), 2)
print(factorial(3), 6)
print(factorial(13), 6)

File: PHP/7php 1/factorial.php
<?php
/*
|--------------------------------------------------------------------------
| Factorial
|--------------------------------------------------------------------------
|
*/

function factorial(int $n) {
   if ($n < 0 || $n > 12) throw new RangeException();
   return $n ? $n * factorial($n - 1) : 1;
}


function factorial(int $n): int {
   if($n < 0 || $n > 12) throw new RangeException;
   return $n ? array_product(range(1, $n)) : 1;
}


function factorial(int $n): int {
   if ($n < 0 || $n > 12) {
     throw new \RangeException;
   }
   $res = 1;
   for ($i = 1; $i <= $n; $i += 1) $res *= $i;
   return $res;
}


print_r(factorial(0) . ' ' . 1 . '
');
print_r(factorial(1) . ' ' . 1 . '
');
print_r(factorial(2) . ' ' . 2 . '
');
print_r(factorial(3) . ' ' . 6 . '
');